src/ephemeral.rs

```diff
@@ -329,23 +328,22 @@ pub(crate) async fn start_ephemeral_timers_msgids(
     msg_ids: &[MsgId],
 ) -> Result<()> {
     let now = time();
-    let count = context
+    let should_interrupt =
+    context
         .sql
-        .execute(
-            &format!(
-                "UPDATE msgs SET ephemeral_timestamp = ? + ephemeral_timer
-         WHERE (ephemeral_timestamp == 0 OR ephemeral_timestamp > ? + ephemeral_timer) AND ephemeral_timer > 0
-         AND id IN ({})",
-                sql::repeat_vars(msg_ids.len())
-            ),
-            rusqlite::params_from_iter(
-                std::iter::once(&now as &dyn crate::sql::ToSql)
-                    .chain(std::iter::once(&now as &dyn crate::sql::ToSql))
-                    .chain(params_iter(msg_ids)),
-            ),
-        )
-        .await?;
-    if count > 0 {
+        .transaction(move |transaction| {
+            let mut should_interrupt = false;
+            for msg_id in msg_ids {
+                should_interrupt |= transaction.execute(
```

Comment:
Still, preparing a statement may improve performance. At least it shouldn't reduce it